apps/src/lab2/views/components/Instructions/NavigationArea.tsx

```diff
@@ -112,28 +109,52 @@ const NavigationArea: React.FC<NavigationAreaProps> = ({
     }
   }, [validationMessage, isRunning]);
 
-  const canShowButton = useMemo(() => {
+  const continueButtonIsEnabled = useMemo(() => {
     if (isPredictLevel) {
       return predictResponseSubmitted;
-    } else if (submittable && hasSubmitted) {
-      return true;
```

Comment:
I think we will still need this condition...if you're revisiting a level you've already submitted (in Music Lab for example) you would have not passed validation until you click run again. In that case, you still want to be able to see the Unsubmit button even before running or editing.